services-core/src/gas_price/priority.rs

```diff
@@ -17,10 +39,21 @@ impl PriorityGasPrice {
         T: Fn(&'a dyn GasPriceEstimating) -> F,
         F: Future<Output = Result<f64>>,
     {
-        for estimator in &self.estimators {
-            match operation(estimator.as_ref()).await {
-                Ok(result) => return Ok(result),
-                Err(err) => log::error!("gas estimator failed: {:?}", err),
+        const ORDERING: Ordering = Ordering::SeqCst;
+        for (i, estimator) in self.estimators.iter().enumerate() {
+            match operation(estimator.estimator.as_ref()).await {
+                Ok(result) => {
+                    estimator.errors_in_a_row.store(0, ORDERING);
+                    return Ok(result);
+                }
+                Err(err) => {
+                    let num_errors = estimator.errors_in_a_row.fetch_add(1, ORDERING) + 1;
+                    if num_errors < LOG_ERROR_AFTER_N_ERRORS {
+                        log::warn!("gas estimator {} failed: {:?}", i, err);
```

Comment:
nit: Could we maybe make GasEstimator implement a display method to print which estimator fails? Printing the index is already a nice improvement but still requires lookup of the current config.

Reply:
I think this is a great idea, but I would just use `Debug` for `#[derive(Debug)]` to do the work for us.

Reply:
Agree it would be nice if it had the name but I don't think looking up the configuration is too much work for a rare occurrence like this either. If we could get the name without any drawbacks then I would definitely do it now but in this case I'm not sure what the best way to do that is. Like Nick said I don't want the trait to require Debug.

Reply:
Could we add a `fn name(&self) -> &str` to the trait? Still feels a bit icky...